src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/Harness/Listeners.cs

```diff
@@ -348,82 +346,76 @@ public override void EventSourceCommand(string eventSourceName, EventCommand com
             };
         }
 
-        public override void Dispose()
+#if netcoreapp
+        private void mListenerEventSourceCreated(object sender, EventSourceCreatedEventArgs eventSource)
         {
-            _listener.Dispose();
-        }
+			_onEventSourceCreated?.Invoke(eventSource.EventSource);
+		}
+#endif
 
-        #region private
-        private void DoCommand(EventSource source, EventCommand command, FilteringOptions options)
+        private void mListenerEventWritten(object sender, EventWrittenEventArgs eventData)
         {
-            if (command == EventCommand.Enable)
-                _listener.EnableEvents(source, options.Level, options.Keywords, options.Args);
-            else if (command == EventCommand.Disable)
-                _listener.DisableEvents(source);
-            else
-                throw new NotImplementedException();
+            OnEvent(new EventListenerEvent(eventData));
         }
 
         private class HelperEventListener : EventListener
         {
-            public HelperEventListener(EventListenerListener forwardTo) { _forwardTo = forwardTo; }
-            protected override void OnEventWritten(EventWrittenEventArgs eventData)
-            {
-#if false // TODO: EventListener events are not enabled in coreclr. GitHub issue #4865
-                base.OnEventWritten(eventData);
-#endif // false
+            private readonly EventListenerListener _forwardTo;
 
-                if (_forwardTo != null && _forwardTo.OnEvent != null)
-                {
-                    _forwardTo.OnEvent(new EventListenerEvent(eventData));
-                }
+            public HelperEventListener(EventListenerListener forwardTo)
+            {
+                _forwardTo = forwardTo;
             }
 
             protected override void OnEventSourceCreated(EventSource eventSource)
             {
                 base.OnEventSourceCreated(eventSource);
 
-                if (_forwardTo != null && _forwardTo._onEventSourceCreated != null)
-                    _forwardTo._onEventSourceCreated(eventSource);
+                _forwardTo?._onEventSourceCreated?.Invoke(eventSource);
             }
 
-            private EventListenerListener _forwardTo;
+            protected override void OnEventWritten(EventWrittenEventArgs eventData)
+            {
+#if netcoreapp
+                // EventSource:L4375 OnEventWritten is internal protected. Can't call...
```

Comment:
this line is definitely problematic... Not sure how to solve it with the current architecture (internal protected vs packaged)...